File: data/transcoder_evaluation_gfg/c/COUNT_SUBSTRINGS_WITH_SAME_FIRST_AND_LAST_CHARACTERS.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

int f_gold ( char s [] ) {
  int result = 0;
  int n = strlen(s);
  for ( int i = 0;
  i < n;
  i ++ ) for ( int j = i;
  j < n;
  j ++ ) if ( s [ i ] == s [ j ] ) result ++;
  return result;
}


int f_filled ( char s [] ) {}

int main(void) {
    int n_success = 0;
    char param0[][100] = {"LZIKA","0556979952","110010","kGaYfd","413567670657","01001","EQPuFa","48848378","110","PLehNeP"};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/go/COUNT_SUBSTRINGS_WITH_SAME_FIRST_AND_LAST_CHARACTERS.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(s []byte) int {
	var (
		result int = 0
		n      int = int(len(s))
	)
	for i := int(0); i < n; i++ {
		for j := int(i); j < n; j++ {
			if s[i] == s[j] {
				result++
			}
		}
	}
	return result
}
//TOFILL
func main() {
	var (
		n_success int           = 0
		param0    [10][]byte = [10][]byte{func() []byte {
			var t []byte
			copy(t[:], ([]byte)("LZIKA"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("0556979952"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("110010"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("kGaYfd"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("413567670657"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("01001"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("EQPuFa"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("48848378"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("110"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("PLehNeP"))
			return t
		}()}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i][:]) == f_gold(param0[i][:]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}
